rnn_speech_recognition/pytorch/train.py

```diff
@@ -14,464 +13,562 @@
+    logging.log_event(logging.constants.SUBMISSION_ORG, value='my-organization')
+    logging.log_event(logging.constants.SUBMISSION_DIVISION, value=logging.constants.CLOSED) # closed or open
+    logging.log_event(logging.constants.SUBMISSION_STATUS, value=logging.constants.ONPREM) # on-prem/cloud/research
+    logging.log_event(logging.constants.SUBMISSION_PLATFORM, value='my platform')
 
-    multi_gpu = torch.distributed.is_initialized()
+    logging.log_end(logging.constants.INIT_STOP)
     if multi_gpu:
-        print_once("DISTRIBUTED TRAINING with {} gpus".format(torch.distributed.get_world_size()))
+        torch.distributed.barrier()
+    logging.log_start(logging.constants.RUN_START)
+    if multi_gpu:
+        torch.distributed.barrier()
+
+    print_once('Setting up datasets...')
+    (
+        train_dataset_kw,
+        train_features_kw,
+        train_splicing_kw,
+        train_specaugm_kw,
+    ) = config.input(cfg, 'train')
+    (
+        val_dataset_kw,
+        val_features_kw,
+        val_splicing_kw,
+        val_specaugm_kw,
+    ) = config.input(cfg, 'val')
+
+    logging.log_event(logging.constants.DATA_TRAIN_MAX_DURATION,
+                      value=train_dataset_kw['max_duration'])
+    logging.log_event(logging.constants.DATA_SPEED_PERTURBATON_MAX,
+                      value=train_dataset_kw['speed_perturbation']['max_rate'])
+    logging.log_event(logging.constants.DATA_SPEED_PERTURBATON_MIN,
+                      value=train_dataset_kw['speed_perturbation']['min_rate'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_FREQ_N,
+                      value=train_specaugm_kw['freq_masks'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_FREQ_MIN,
+                      value=train_specaugm_kw['min_freq'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_FREQ_MAX,
+                      value=train_specaugm_kw['max_freq'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_TIME_N,
+                      value=train_specaugm_kw['time_masks'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_TIME_MIN,
+                      value=train_specaugm_kw['min_time'])
+    logging.log_event(logging.constants.DATA_SPEC_AUGMENT_TIME_MAX,
+                      value=train_specaugm_kw['max_time'])
+    logging.log_event(logging.constants.GLOBAL_BATCH_SIZE,
+                      value=batch_size * world_size * args.grad_accumulation_steps)
+
+    tokenizer_kw = config.tokenizer(cfg)
+    tokenizer = Tokenizer(**tokenizer_kw)
+
+    class PermuteAudio(torch.nn.Module):
+        def forward(self, x):
+            return (x[0].permute(2, 0, 1), *x[1:])
+
+    train_augmentations = torch.nn.Sequential(
+        train_specaugm_kw and features.SpecAugment(optim_level=args.amp, **train_specaugm_kw) or torch.nn.Identity(),
+        features.FrameSplicing(optim_level=args.amp, **train_splicing_kw),
+        features.FillPadding(optim_level=args.amp, ),
```

Comment:
I am not sure if this is `FillPadding` is required since a padding is already done as part of the DALI pipeline here: 
[DALI padding](https://github.com/mwawrzos/training/blob/rnnt/rnn_speech_recognition/pytorch/common/data/dali/pipeline.py#L118)

Reply:
Yeah, you are right. It is left after the old pipeline. Let me make sure it can be safely removed, and I can update the code.